sphinx_gallery/gen_gallery.py

```diff
@@ -79,6 +79,7 @@ def __call__(self, gallery_conf, script_vars):
     'thumbnail_size': (400, 280),  # Default CSS does 0.4 scaling (160, 112)
     'min_reported_time': 0,
     'binder': {},
+    'promote_jupyter_magic': False,
```

Comment:
Maybe some type checking to ensure the `promote_jupyter_magic` input is correct?